test/integration/error_output.cc

```diff
@@ -219,3 +222,103 @@ TEST(ErrorOutput, ElementErrorOutput)
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+
+  sdf::ParserConfig parserConfig;
+  parserConfig.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+  parserConfig.SetFindCallback(findFileCb);
+  sdf::readString(stream.str(), parserConfig, sdfParsed, errors);
+  EXPECT_TRUE(errors.empty());
+
+  sdf::Model model;
+  errors = model.Load(sdfParsed->Root()->GetElement("model"), parserConfig);
+  ASSERT_EQ(errors.size(), 7u);
+  EXPECT_NE(std::string::npos, errors[0].Message().find(
+    "Non-unique name[common_name] detected 7 times in XML children of model"
+    " with name[test_model]."));
+  EXPECT_NE(std::string::npos, errors[1].Message().find(
+    "link with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[2].Message().find(
+    "Link with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this link."));
+  EXPECT_NE(std::string::npos, errors[3].Message().find(
+    "joint with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[4].Message().find(
+    "Joint with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this joint."));
+  EXPECT_NE(std::string::npos, errors[5].Message().find(
+    "frame with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[6].Message().find(
+    "Frame with name [common_name] in model with name [test_model] has a name"
+    " collision. Please rename this frame."));
+  errors.clear();
+
+  //Set SDF version to someting lower than 1.7 for extra errors
+  sdfParsed->SetOriginalVersion("1.6.0");
+  sdf::Model second_model;
+  sdfParsed->Root()->GetElement("model")->SetOriginalVersion("1.2.0");
+  errors = second_model.Load(sdfParsed->Root()->GetElement("model"), parserConfig);
+
+  ASSERT_EQ(errors.size(), 7u);
+  EXPECT_NE(std::string::npos, errors[0].Message().find(
+    "Non-unique name[common_name] detected 7 times in XML children of model"
+    " with name[test_model]."));
+  EXPECT_NE(std::string::npos, errors[1].Message().find(
+    "link with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[2].Message().find(
+    "Link with name [common_name] in model with name [test_model] has a name"
+    " collision, changing link name to [common_name_link]."));
+  EXPECT_NE(std::string::npos, errors[3].Message().find(
+    "joint with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[4].Message().find(
+    "Joint with name [common_name] in model with name [test_model] has a name"
+    " collision, changing joint name to [common_name_joint]."));
+  EXPECT_NE(std::string::npos, errors[5].Message().find(
+    "frame with name[common_name] already exists."));
+  EXPECT_NE(std::string::npos, errors[6].Message().find(
+    "Frame with name [common_name] in model with name [test_model] has a name"
+    " collision, changing frame name to [common_name_frame]."));
+
```

Comment:
Can you add a test where the error policy is set to the default to make sure warnings are not reported in `errors`?

Reply:
Done in: https://github.com/gazebosim/sdformat/pull/1122/commits/cb8099b458d95fa9cc05fd6a843fa064fe97d459.